tutorials/realm/MachineModel.md

```diff
@@ -0,0 +1,134 @@
+provides better performance for processors on the same socket.
+- `Z_COPY_MEM` represents Zero-Copy memory visible to all CPUs within a node and one or more GPUs.
+It can be specified by `-ll:zsize`.
+- `GPU_FB_MEM` represents framebuffer memory for a particular GPU.
+It can be specified by `-ll:fsize`.
+- `GPU_MANAGED_MEM` represents managed memory that can be cached by either host or GPU.
+It can be specified by `-ll:msize`.
+- `GPU_DYNAMIC_MEM` represents dynamically-allocated framebuffer memory for a particular GPU.
+Its size is not fixed, but its maximum size can be specified by `-cuda:dynfb_max`.
+- `DISK_MEM` represents disk memory visible to all processors on a node.
+It can be specified by `-ll:dsize`.
+- `HDF_MEM` and `FILE_MEM` represent HDF and file memory visible to all processors on a node, respectively. 
+They do not have memory space, so their sizes are always 0.
+These I/O related memories allow users to create instances for I/O operations. 
+
+For a list of all the memories kind supported by Realm, please refer to [Full Memory Kind](#full-mem-kind).
+
+## Affinity
+
+Realm provides `ProcessorMemoryAffinity` and `MemoryMemoryAffinity` to query the affinity information
+between processors and memories. In this example, we use `ProcessorMemoryAffinity` to retrieve the information,
+including latency and bandwidth between a pair of memory and processor.
+```c++
+std::vector<Machine::ProcessorMemoryAffinity> pm_affinity;
+machine.get_proc_mem_affinity(pm_affinity, p, m, true/*local_only*/);
+unsigned bandwidth = pm_affinity[0].bandwidth;
+unsigned latency = pm_affinity[0].latency;
+``` 
+
+## ID
+
+Realm ID is a 64-bit value that uniquely encodes both the type of the referred-to Realm object and its identity.
+Once we convert an ID into a hexadecimal number, it can be decoded. The following is an example of a processor ID 
+and a memory ID:
+```
+Processor ID 1d00010000000001 is CPU.
+System Memory ID 1e00010000000000 has 0 KB, bandwidth 100, latency 5.
+```
+The highest two digits (8 bits) are used to tell the type of an ID, e.g., `1d` represents Processor and `1e` represents Memory. 
+The next four digits (16 bits) are used to tell the owner node of an ID, e.g., `0001` means the processor/memory is on
+node 1. The last two/three digits (8/12 bits) are used tell the local index of a Memory/Processor ID, e.g., `01` means
+the cpu index is 1 while `00` means the memory index is 0. 
+Besides processor and memory, the ID of other Realm objects (e.g., Event and etc.) can also be decoded. 
+For a complete introduction to Realm ID, please refer to the [ID header file](#id-header-file).
+
+Realm provides `is_TYPE` functions to test the type of an ID, e.g. in this example, `is_processor` is used to check if an ID 
+is a processor.
+
+## References
+
+<div id="full-proc-kind"></div>
+[1]: [Full Processor Kind](https://github.com/StanfordLegion/legion/blob/stable/runtime/realm/realm_c.h#L45)
```

Comment:
When you're using literal `[` as in `[1]`, you need to escape the opening `[` to make sure it doesn't accidentally turn into a link:

```
\[1]: ...
```

Reply:
fixed